pkg/metastore/glue/metastore_client.go

```diff
@@ -16,25 +16,31 @@ import (
 const MaxParts = 1000 // max possible 1000
 
 type MSClient struct {
-	client    glueiface.GlueAPI
-	catalogID string
+	client          glueiface.GlueAPI
+	catalogID       string
+	baseLocationURI string
+}
+
+func (g *MSClient) GetDBLocation(dbName string) string {
+	return fmt.Sprintf("%s/%s", g.baseLocationURI, dbName)
 }
 
 func (g *MSClient) NormalizeDBName(db string) string {
 	return db
 }
 
-func NewMSClient(cfg *aws.Config, catalogID string) (metastore.Client, error) {
+func NewMSClient(cfg *aws.Config, catalogID, baselLocationURI string) (metastore.Client, error) {
 	sess := session.Must(session.NewSession(cfg))
 	sess.ClientConfig("glue")
 	gl := glue.New(sess)
 	return &MSClient{
-		client:    gl,
-		catalogID: catalogID,
+		client:          gl,
+		catalogID:       catalogID,
+		baseLocationURI: baselLocationURI,
```

Comment:
I think this is the point where we need to trim right '/' or make sure we have '/' on the right (same for the other client).
Check GetDBLocation - in one case we trim and the other we don't. if we know the state of the member - we don't need to assume on each get